docs/connector-development/config-based/understanding-the-yaml-file/incremental-syncs.md

```diff
@@ -0,0 +1,171 @@
+        additionalProperties: true
+  MinMaxDatetime:
+    description: Compares the provided date against optional minimum or maximum times. The max_datetime serves as the ceiling and will be returned when datetime exceeds it. The min_datetime serves as the floor
+    type: object
+    required:
+      - type
+      - datetime
+    properties:
+      type:
+        type: string
+        enum: [MinMaxDatetime]
+      datetime:
+        type: string
+      datetime_format:
+        type: string
+        default: ""
+      max_datetime:
+        type: string
+      min_datetime:
+        type: string
+      $parameters:
+        type: object
+        additionalProperties: true
+```
+
+Example:
+
+```yaml
+incremental_sync:
+  type: DatetimeBasedCursor
+  start_datetime: "2022-01-01T00:00:00.000000+0000"
+  end_datetime: "2022-01-05T00:00:00.000000+0000"
+  datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+  cursor_granularity: "PT0.000001S"
+  step: "P1D"
+```
+
+will result in the datetime partition windows in the example mentioned earlier.
+
+### Lookback Windows
+
+The `DatetimeBasedCursor` also supports an optional lookback window, specifying how many days before the start_datetime to read data for.
+
+```yaml
+incremental_sync:
+  type: DatetimeBasedCursor
+  start_datetime: "2022-02-01T00:00:00.000000+0000"
+  end_datetime: "2022-03-01T00:00:00.000000+0000"
+  datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+  cursor_granularity: "PT0.000001S"
+  lookback_window: "P31D"
+  step: "P1D"
+```
+
+will read data from `2022-01-01` to `2022-03-01`.
+
+The stream partitions will be of the form `{"start_date": "2021-02-01T00:00:00.000000+0000", "end_date": "2021-02-02T23:59:59.999999+0000"}`
+The stream partitions' field names can be customized through the `stream_state_field_start` and `stream_state_field_end` parameters.
+
+The `datetime_format` can be used to specify the format of the start and end time. It is [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) by default.
+
+The Stream's state will be derived by reading the record's `cursor_field`.
+If the `cursor_field` is `updated_at`, and the record is `{"id": 1234, "created": "2021-02-02T00:00:00.000000+0000"}`, then the state after reading that record is `"updated_at": "2021-02-02T00:00:00.000000+0000"`. [^1]
+
+Note that all durations are expressed as [ISO 8601 durations](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+
+### Filtering according to Cursor Field
+
+If an API supports filtering data based on the cursor field, the `start_time_option` and `end_time_option` parameters can be used to configure this filtering.
+For instance, if the API supports filtering using the request parameters `created[gte]` and `created[lte]`, then the component can specify the request parameters as
+
+```yaml
+incremental_sync:
+  type: DatetimeCursorBased
+  <...>
+  start_time_option:
+    type: RequestOption
+    field_name: "created[gte]"
+    inject_into: "request_parameter"
```

Comment:
I find it interesting that `request_parameter` seems like a very HTTP centric term but `incremental_sync` is now at the root of the stream and not under a HTTP-centric concept. Should we be afraid of how it might evolve when we will add things like gRPC or whatever?

Reply:
As agreed, this is the less problematic answer between this and putting the information on the requester level. Therefore, we'll be resolving this